src/dbnode/storage/index_test.go

```diff
@@ -432,8 +433,10 @@ func verifyFlushForShards(
 		mockBlock.EXPECT().EvictMutableSegments().Return(nil)
 	}
 	require.NoError(t, idx.WarmFlush(mockFlush, dbShards))
-	require.Equal(t, persistClosedTimes, numBlocks)
-	require.Equal(t, persistCalledTimes, numBlocks)
+	require.True(t, xclock.WaitUntil(func() bool {
+		return persistClosedTimes == numBlocks &&
+			persistCalledTimes == numBlocks
+	}, 5*time.Second))
```

Comment:
guessing 5s is enough time?

Reply:
yea, probably don't even need to wait that long the test normally runs sub .01s